### PR TITLE
[flutter_tools] even more test fixes

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Fix tests broken under tester

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/device_test.dart [hunk 1]
```diff
@@ -25,6 +26,8 @@ void main() {
       final List<Device> devices = <Device>[device1, device2, device3];
       final DeviceManager deviceManager = TestDeviceManager(devices);
       expect(await deviceManager.getDevices(), devices);
+    }, overrides: <Type, Generator>{
+      Artifacts: () => Artifacts.test(),
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/device_test.dart:30]: Hitting real artifacts, failing if they did not exist

File: packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart [hunk 2]
```diff
@@ -791,28 +792,36 @@ void main() {
   });
 
   testUsingContext('Correct flutter runner', () async {
-    expect(globals.artifacts.getArtifactPath(
+    final MockCache cache = MockCache();
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/fuchsia/fuchsia_device_test.dart:795]: Hitting real cache, failing if it didn't exist.

File: packages/flutter_tools/test/general.shard/hot_test.dart [hunk 3]
```diff
@@ -155,11 +159,13 @@ void main() {
     when(mockDevFs.lastCompiled).thenReturn(DateTime.now());
 
     setUp(() {
-      mockArtifacts = MockLocalEngineArtifacts();
-      when(mockArtifacts.getArtifactPath(Artifact.flutterPatchedSdkPath)).thenReturn('some/path');
+      fileSystem = MemoryFileSystem.test();
     });
 
     testUsingContext('Does not hot restart when device does not support it', () async {
+      fileSystem.file(globalPackagesPath)
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/hot_test.dart:166]: hitting real file system, failing if a .packages didn't exist

File: packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart [hunk 4]
```diff
@@ -52,7 +51,6 @@ void main() {
 
   String podsIsInCustomDir({String cocoapodsReposDir}) {
     cocoapodsReposDir ??= fileSystem.path.join(
-      globals.fsUtils.homeDirPath,
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/macos/cocoapods_test.dart:55]: This test apparently doesn't rely on this value, avoid looking up potentially null home dir path

File: packages/flutter_tools/test/general.shard/web/devfs_web_test.dart [hunk 5]
```diff
@@ -43,7 +43,9 @@ void main() {
   MockHttpServer mockHttpServer;
 
   setUpAll(() async {
-    packages = await loadPackageConfigUri(Uri.base.resolve('.packages'));
+    packages = PackageConfig(<Package>[
```
Comment (jonahwilliams) [inline packages/flutter_tools/test/general.shard/web/devfs_web_test.dart:46]: This fails if .packages doesn't exist

**Review by jmagman — APPROVED** [review]

**Pull request merged**